src/jdk.compiler/share/classes/com/sun/tools/javac/util/RichDiagnosticFormatter.java

```diff
@@ -83,8 +83,37 @@ public class RichDiagnosticFormatter extends
     /* type/symbol printer used by this formatter */
     private RichPrinter printer;
 
+    private static class WhereClauses {
+        private final Map<WhereClauseKind, Map<Type, JCDiagnostic>> whereClauses;
+
+        WhereClauses() {
+            Map<WhereClauseKind, Map<Type, JCDiagnostic>> whereClauses = new LinkedHashMap<>();
```

Comment:
Looks like a great candidate for `EnumMap`.

Reply:
Thanks, fixed. (The outer map was an EnumMap before, this was an oversight.)